Lesson 11: Object exercises (#14)
ReDI-School/js-berlin-2021-spring: lesson11.md

```js
let person = {
  name: "Anne",
  sayHello: function () {
    console.log("Hi, my name is " + this.name);
  },
};
person.sayHello();
```
<!-- .element: class="fragment" -->

---

# More practice with Objects

---

### Exercise 1

Given the person:

```js
let person = {
  name: "Anne",
  age: 34
};
```

Create the function `celebrateBirthday`, which takes a `person` object, prints a birthday message, stating the person's name and age. After it is called, the person's `age` property should be updated.

```js
celebrateBirthday(person);
// -> "Happy birthday! Anne is now 35!
console.log(person.birthday); // -> 35
```

---

### Exercise 2

Using the `person` object from the previous example, give it a method (function) called `canRentCar` that returns `true` if they are old enough to rent a car (must be over 18 years old), and `false` otherwise.

```js
// `person` is "Anne" age 34 from exercise 1
console.log(person.canRentCar()); // -> true
```

Create another person under the age requirement, and verify that your method returns false for them.

Bonus: Use the function from exercise 1 to age the younger person until they are old enough to rent a car.

---

### Exercise 3

> Create a function `determineOlderPerson` that takes two person objects and prints who is the old person of the two, and by how many years difference.

Author: ```suggestion
Create a function `determineOlderPerson` that takes two person objects and prints who is the older person of the two, and the age difference between them.
```
